### ISSUE TITLE
Training on a structured dataset as json

### ISSUE THREAD

**Issue by fischhaber** [oobabooga/text-generation-webui#3536]
I want to train a LoRA with my own dataset. The dataset is as follows:
[
      {
        "question": "abc",
        "call": "def",
        "answer": "ghi"
      },
      {
        "question": "jkl",
        "call": "mno",
        "answer": "pqr"
      },
      {
        "question": "stu",
        "call": "vwx",
        "answer": "yza"
      }
]
It is not the vicuna format because it has no "instruction" and "output". What should I change that this dataset will work? Should I create a new format?

**Comment by bombel28** [comment]
Yes, just create a new template. I managed to ask chatgpt to create a template for oasst1 dataset. But I did not test it. Copy and paste all of the three format templates, show it your favourite gpt model and then show your dataset to create a new format template. I also managed to get a complete script for merging Lora file with the model, and it works fine. It loads the model in 8 bit to save vram. It is very very easy, chatgpt knows really much about the transformers library.

**Comment by bombel28** [comment]
Meanwhile... My own format template is not working :-D
But I try to help you... maybe, this could work:
{
    "question,call,answer": "User: %question%\nAssistant: %call%\n%answer%"
}

**Comment by fischhaber** [comment]
Oh ok, thank you for your answers! I had tried it with 
{
"question,call,answer": "USER: %question%\n\nASSISTANT: %call%\n\nASSISTANT:%answer%"
}

Do you believe that this can make a difference? 

The call should be a python call. After training it can not give the right python call back and it can not give the right answer to my question. Do you have any suggestions?
